### PR TITLE
JS: Add models for `webix`

### PR DESCRIPTION
**Pull request by jorgectf**
This PR adds modeling for the [Webix](https://docs.webix.com/) framework.

cc @kwstubbs

### CODE REVIEW

**Comment by asgerf** [comment]
LGTM, just need an update to the test expectations for `HeuristicSourceCodeInjection.ql`, which resides in the same folder as the regular code injection test.

File: javascript/ql/lib/semmle/javascript/security/dataflow/CodeInjectionCustomizations.qll [hunk 1]
```diff
@@ -312,6 +312,13 @@ module CodeInjection {
     }
   }
 
+  /**
+   * A value interpreted as code by the `webix` library.
+   */
+  class WebixExec extends Sink {
+    WebixExec() { this = DataFlow::moduleImport("webix").getAMemberCall("exec").getArgument(0) }
```
Comment (asgerf) [inline javascript/ql/lib/semmle/javascript/security/dataflow/CodeInjectionCustomizations.qll:319]:
```suggestion
    WebixExec() { this = API::moduleImport("webix").getMember("exec").getParameter(0).asSink() }
```

File: javascript/ql/lib/semmle/javascript/security/dataflow/CodeInjectionCustomizations.qll [hunk 2]
```diff
@@ -419,6 +426,23 @@ module CodeInjection {
     }
   }
 
+  /**
+   * A value interpreted as a template by the `webix` library.
+   */
+  class WebixTemplateSink extends TemplateSink {
+    WebixTemplateSink() {
+      this = DataFlow::moduleImport("webix").getAMemberCall("ui").getOptionArgument(0, "template")
```
Comment (asgerf) [inline javascript/ql/lib/semmle/javascript/security/dataflow/CodeInjectionCustomizations.qll:434]:
```suggestion
      this = API::moduleImport("webix").getMember("ui").getParameter(0).getMember("template").asSink()
```

File: javascript/ql/lib/semmle/javascript/security/dataflow/CodeInjectionCustomizations.qll [hunk 3]
```diff
@@ -419,6 +426,23 @@ module CodeInjection {
...
+      or
+      this.asExpr() =
+        DataFlow::moduleImport("webix")
+            .getAMemberCall("ui")
+            .getOptionArgument(0, "template")
+            .asExpr()
+            .(Function)
+            .getAReturnedExpr()
```
Comment (asgerf) [inline javascript/ql/lib/semmle/javascript/security/dataflow/CodeInjectionCustomizations.qll:442]:
```suggestion
      this.asExpr() =
        API::moduleImport("webix")
            .getMember("ui")
            .getParameter(0)
            .getMember("template")
            .getReturn()
            .asSink()
```

**Review by asgerf — COMMENTED** [review]
Some optional comments for using API graphs instead of local data flow. Note that they're written off the top of the my head and I may have made mistakes.

**Comment by jorgectf** [comment]
> Some optional comments for using API graphs instead of local data flow. Note that they're written off the top of the my head and I may have made mistakes.

Thanks @asgerf! Done ✅

**Comment by Kwstubbs** [comment]
@asgerf I noticed that webix could be used in the form of 
```javascript 
<script src="../../codebase/webix.js" type="text/javascript" charset="utf-8"></script>
<script>let whatever = webix.extend(object1, object_with_usercontrolled_values) </script> 
```
Would CodeQL be able to identify this script tag contents as a sink? We would like to support this use case as well if possible. Thanks!

**Review by asgerf — COMMENTED** [review]
> Would CodeQL be able to identify calls inside script tags (within an HTML file)? We would like to support this use case as well if possible. Thanks!

Yes, we'd usually use `DataFlow::globalVarRef("webix")` to find uses of the global variable `webix`, see the inline suggestion.

It's slightly cumbersome to get that working with API graphs (we're working on simplifying it). A good example of how to do this can be found in the D3 model: https://github.com/github/codeql/blob/36d86787e6fbbb158a148f93eb63f829134fd6b1/javascript/ql/lib/semmle/javascript/frameworks/D3.qll#L24

**Comment by jorgectf** [comment]
Nice catch @Kwstubbs, I have adapted the modeling as per https://github.com/github/codeql/pull/13529#pullrequestreview-1502908960 @asgerf suggestion

File: javascript/ql/lib/semmle/javascript/frameworks/Webix.qll [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  private class WebixGlobalEntry extends API::EntryPoint {
+    WebixGlobalEntry() { this = "WebixGlobalEntry" }
+
+    override DataFlow::SourceNode getASource() { result = DataFlow::globalVarRef("webix") }
+  }
+
+  /** Gets a reference to the Webix package. */
+  API::Node webix() {
+    result = API::moduleImport("webix") or
+    result.asSource() = DataFlow::moduleImport("webix") or
```
Comment (asgerf) [inline javascript/ql/lib/semmle/javascript/frameworks/Webix.qll:21]:
```suggestion
```
This line seems unnecessary. Was there some reason for adding it?

**Review by asgerf — COMMENTED** [review]
One final comment. I'll start an evaluation of the PR in the meantime.

File: javascript/ql/lib/semmle/javascript/frameworks/Webix.qll (hunk 4 above)
Comment (jorgectf) [reply javascript/ql/lib/semmle/javascript/frameworks/Webix.qll:21]: You are right! Thanks :D

**Review by asgerf — APPROVED** [review]

**Pull request merged**